### PR TITLE
feat(APP-1001): Filter plugin actions in action composer by DAO permissions

### PR DESCRIPTION
**Pull request by milosh86**
# Description

Filters OSx-related plugin actions out of the action builder when the DAO does not hold the required permission to execute them, so users are no longer offered actions that would fail on execution.

Each plugin action item can now declare a `requiredPermissionId` (keccak256 permission id). When building the DAO action list, `actionComposerUtils` keeps only the plugin items whose required permission is granted to the DAO, then prunes any plugin groups left without items.

Details:
- Add `requiredPermissionId?: string` to `IActionComposerInputItem`. Matching is by permission id and target contract (`defaultValue.to` as `where`); `who` is checked against DAO address.
- `filterItemsByPermission` fails closed while permissions are still loading (`undefined`), always keeps items without a `requiredPermissionId`, and throws (`invariant`) when a permission-tagged item has no resolvable target address (misconfiguration guard).
- `pruneEmptyGroups` drops plugin groups whose actions were all filtered out.
- Tag the relevant plugin actions with their permission ids:
  - Multisig: add/remove members and update settings → `UPDATE_MULTISIG_SETTINGS_PERMISSION`
  - Token: mint → `MINT_PERMISSION`, update voting settings → `UPDATE_VOTING_SETTINGS_PERMISSION`
  - Lock-to-vote: update settings → `UPDATE_SETTINGS_PERMISSION`
  - Metadata: set metadata → `SET_METADATA_PERMISSION`
- Add unit tests for `getDaoActions` covering permission filtering and empty-group pruning.

## Type of Change

- [ ] Major: Breaking change (change that would cause existing functionality to not work as expected)
- [x] Minor: Feature (non-breaking change which adds new functionality)
- [ ] Patch: Enhancement (non-breaking change to an existing feature)
- [ ] Patch: Bug fix (non-breaking change which fixes an issue)

## Developer Checklist:

- [ ] Manually smoke tested the functionality in a preview or locally
- [ ] Confirmed there are no new warnings or errors in the browser console
- [ ] (For User Stories only) Double-checked that all Acceptance Criteria are satisfied
- [x] Confirmed there are no new warnings on automated tests
- [ ] Merged and published any dependent changes in downstream modules
- [x] Selected the correct base branch
- [x] Commented the code in hard-to-understand areas
- [x] Followed the code style guidelines of this project
- [ ] Reviewed that the Files Changed in Github's UI reflect my intended changes
- [ ] Confirmed the pipeline checks are not failing

## Review Checklist:

- [ ] (For User Stories only) Tested in a preview or locally that all Acceptance Criteria are satisfied
- [ ] Confirmed that changes follow the code style guidelines of this project

### CODE REVIEW

File: apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts [hunk 1]
```diff
@@ -51,15 +56,83 @@ class ActionComposerUtils {
...
+                where != null && where !== '',
+                `filterItemsByPermission: action "${item.id}" declares a requiredPermissionId but has no target address.`,
+            );
+
+            return permissions.some(
+                (permission) =>
+                    permission.permissionId.toLowerCase() ===
+                        requiredPermissionId.toLowerCase() &&
+                    addressUtils.isAddressEqual(permission.whereAddress, where),
+            );
```
Comment (thekidnamedkd) [inline apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts:126]: We're matching `permissionId` + `where` only, but APP-1001 keys executability on the DAO holding the permission as `who`. So the Alchemix case (granted to someone, but not the DAO) still shows here and reverts on execution. Closing it means threading `dao.address` through and matching `whoAddress` too. Fine as a follow-up if we'd rather scope it out.

File: apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts [hunk 2]
```diff
@@ -51,15 +56,83 @@ class ActionComposerUtils {
...
+     * with no resolvable target throws, as that indicates a misconfigured action.
+     *
+     * @param items - Plugin action items to filter.
+     * @param permissions - Permissions granted to the DAO, or undefined when not loaded.
+     * @returns The items the DAO is authorized to execute.
+     */
+    private filterItemsByPermission = (
+        items: IActionComposerInputItem[],
+        permissions: IGetDaoActionsParams['permissions'] = [],
+    ): IActionComposerInputItem[] => {
```
Comment (thekidnamedkd) [inline apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts:96]: Minor, non-blocking: the doc says we fail open while loading, but the `= []` default fails closed, so gated actions blink out then back in as permissions page in. If we want fail-open, guarding on `== null` does it (`!length` would also hide them for a DAO with genuinely zero perms):

```suggestion
    private filterItemsByPermission = (
        items: IActionComposerInputItem[],
        permissions: IGetDaoActionsParams['permissions'],
    ): IActionComposerInputItem[] => {
        if (permissions == null) {
            return items;
        }
```

**Review by thekidnamedkd — DISMISSED** [review]
Couple of inline notes, nothing gating from me. Main one's the `who` check APP-1001 asks for.

**Review by thekidnamedkd — COMMENTED** [review]
Reviewed, leaving this as comments. Two notes inline: the `who` check APP-1001 wants (main one, happy for it to be a follow-up), and a minor fail-open/closed doc mismatch. Not gating from me, so it still needs an approver to unblock.

File: apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts (hunk 2 above)
Comment (milosh86) [reply apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts:96]: I can't find the doc mentioning it should fail open. It must be run on some older commit as I refactor fail-open to fail-closed intentionally. If an action has a "required" permission, then it makes sense to show it only when validation pass.

File: apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts (hunk 1 above)
Comment (milosh86) [reply apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts:126]: hey, yes, I initially implemented `who` checks but it added some complexity I didn't like, so I Ieft it only with `where` check. I would rather leave it as an future improvement, but only if needed, since I don't see currently a case where something else is set as `who` other than DAO. Also, the worst case is that action would be visible, but not executable. Let me know if you see some case where this would make sense to check.

File: apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts (hunk 2 above)
Comment (thekidnamedkd) [reply apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts:96]: Ah sorry, not a doc, the PR body here says `filterItemsByPermission` fails open while permissions are loading. If fail-closed is intentional now, code can stay as-is, just wanted to clarify and keep the description aligned.

File: apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts (hunk 1 above)
Comment (thekidnamedkd) [reply apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts:126]: The part I’m stuck on is that APP-1001 frames executability as the DAO address being `who`, while this checks `permissionId + where`. If we’re intentionally scoping this to “some permission exists on the target” for now, can we make that explicit in the PR and track `who` as a follow-up? Or maybe I’m not understanding the distinction?

File: apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts (hunk 2 above)
Comment (milosh86) [reply apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts:96]: yeah, I see now in the PR body 👍  ... anyway, fail-closed was the goal.

File: apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts (hunk 1 above)
Comment (milosh86) [reply apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts:126]: My assumption was that `who` is a DAO in 99.9% of cases in those specific plugin related permissions (which might be a wrong assumption), so it was not worth handling who as  it could be also ANY_ADDRESS.

But yeah, let's better be explicit, I'll add strict DAO expectation for `who`.

Comment (milosh86) [reply apps/app/src/modules/governance/components/actionComposer/actionComposerUtils.ts:126]: added who check..

**Review by thekidnamedkd — APPROVED** [review]
LGTM.
